lib/config/presets/index.ts

```diff
@@ -340,15 +363,67 @@ export async function resolveConfigPresets(
         val as RenovateConfig,
         baseConfig,
         ignorePresets,
-        existingPresets
+        existingPresets,
+        shallowResolve
       );
     }
   }
   logger.trace({ config: inputConfig }, 'Input config');
   logger.trace({ config }, 'Resolved config');
+  if (shallowResolve) {
+    handleExtendsArray(config, unresolvedPresets);
+  }
   return config;
 }
 
+// handleExtendsArray:
+// This function edits extends array presets when shallowResolve is enabled
+// 1. remove resolved external presets from the extends array.
+// 2. clean duplicate presets keys (from different resolution levels).
+function handleExtendsArray(
```

Comment:
Can this be modified to *return* the modified config copy instead of mutating it?

Reply:
Sure will change it